include/aws/common/atomics_gnu_old.inl

```diff
@@ -21,9 +21,16 @@
 #include <stdlib.h>
 
 #if defined(__GNUC__)
-#    if __GNUC__ < 4 || (__GNUC__ == 4 && __GNUC_MINOR__ < 4)
-/* See https://gcc.gnu.org/bugzilla/show_bug.cgi?id=36793 */
-#        error GCC versions before 4.4.0 are not supported
+#    if (__GNUC__ < 4)
+#        error GCC versions before 4.1.2 are not supported
+#    elif (defined(__arm__) || defined(__ia64__)) && (__GNUC__ == 4 && __GNUC_MINOR__ < 4)
+/* See https://gcc.gnu.org/bugzilla/show_bug.cgi?id=36793 Itanium codegen */
+/* https://bugs.launchpad.net/ubuntu/+source/gcc-4.4/+bug/491872 ARM codegen*/
+/* https://gcc.gnu.org/bugzilla/show_bug.cgi?id=42263 ARM codegen */
+#        error GCC versions before 4.4.0 are not supported on ARM or Itanium
+#    elif (defined(__x86_64__) || defined(__i386__)) && (__GNUC__ == 4 && __GNUC_MINOR__ < 1 && __GNUC_PATCHLEVEL__ < 2)
```

Comment:
version check needs fix, 4.1.1, 4.0.8, etc... will pass this check

probably:

M == 4 && ((Minor < 1 || (Minor == 1 && Patch < 2))))

Reply:
:fearful: ahem, nothing to see here.